cmake/information_helpers.cmake

```diff
@@ -1,11 +1,98 @@
+function(filter_generator_expressions INPUT OUTPUT)
+    string(REGEX REPLACE ".+LINK:.+:(.+)>" "\\1"
+                    TMP "${INPUT}")
+    string(REGEX REPLACE ".+\\$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:.+>(.+)" "\\1"
+                    TMP "${TMP}")
+    string(REGEX REPLACE ".+INTERFACE:.+>" "" TMP "${TMP}")
```

Comment:
Where do all of these occur? I don't see them in my output.

Reply:
In my version, 3.20.5, they appear. https://gitlab.kitware.com/cmake/cmake/-/blob/v3.20.5/Modules/FindMPI.cmake#L1175

Which CMake version do you use so that I can test the other problems? I should also browse the different versions to see if there are any more of those annoying generators which they wrap everything within the FindMPI.

Reply:
I am using 3.16.3

Reply:
This should work like expected. I tried a local cmake-3.16.3 install and the extra `SHELL:` should be taken care of. The only thing I couldn't try are the two concurrent builds, but since everything uses generator expressions now I don't think you should see the previous issues anymore?